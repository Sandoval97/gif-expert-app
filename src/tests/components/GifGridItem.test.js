import { shallow } from "enzyme"
import GifGridItem from "../../components/GifGridItem"

describe("Pruebas para el componente GifGridItem", () => {
    let wrapper;

    const imageTitle = "This is a title"
    const url = "https://localhost/image.gif"
    beforeEach(() => {

        wrapper = shallow(<GifGridItem title={imageTitle} url={url} />)
    })
    test("Debe mostrar el componente correctamente", () => {

        expect(wrapper).toMatchSnapshot()
    })

    test("Debe mostrar el title adecuado", () => {

        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(imageTitle)
    })

    test("La imagen debe tener la url pasada como parametro", () => {

        const img = wrapper.find('img')

        expect(img.props().src).toBe(url)
        expect(img.props().alt).toBe("Cargando...")
    })

    test("El contenedor padre debe tener la clase para la animacion fadeIn", () => {

        const div = wrapper.find('div')
        const className = div.prop('className')

        expect(className.includes('animate__fadeIn')).toBe(true)
    })
})