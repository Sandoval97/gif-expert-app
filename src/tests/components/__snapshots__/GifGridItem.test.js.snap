// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para el componente GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Cargando..."
    src="https://localhost/image.gif"
  />
  <p>
    This is a title
  </p>
</div>
`;
